src/utils/word-api.ts

```diff
@@ -0,0 +1,205 @@
+      if (error instanceof OfficeExtension.Error) {
+        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
+      }
+    });
+  }
+
+  getCitationTagForPosition = (
+    citationByIndex: Array<StatefulCitation>,
+    position: number
+  ): string => {
+    return `JABREF-CITATION-${JSON.stringify(citationByIndex[position])}`;
+  };
+
+  getCitationContentControlAtPosition = (
+    contentControls: Word.ContentControlCollection,
+    position: number
+  ): Word.ContentControl => {
+    let pos = 0;
+    let contentControl: Word.ContentControl = null;
+    for (let i = 0, ilen = contentControls.items.length; i < ilen; i += 1) {
+      const { tag } = contentControls.items[i];
+      if (tag.includes("JABREF-CITATION")) {
+        if (pos === position) {
+          contentControl = contentControls.items[i];
+          break;
+        } else {
+          pos += 1;
+        }
+      }
+    }
+    return contentControl;
+  };
+
+  static async getTotalNumberOfCitation(): Promise<number | void> {
+    return Word.run(async (context: Word.RequestContext) => {
+      const { contentControls } = context.document;
+      context.load(contentControls, "tag, length");
+      await context.sync();
+      let length = 0;
+      contentControls.items.forEach((citation) => {
+        const { tag } = citation;
+        if (tag.includes("JABREF-CITATION")) {
+          length += 1;
+        }
+      });
+      return length;
+    }).catch((error) => {
+      console.log(`Error: ${JSON.stringify(error)}`);
+      if (error instanceof OfficeExtension.Error) {
+        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
+      }
+    });
+  }
+
+  static async getPositionOfNewCitation(): Promise<number | void> {
+    return Word.run(async (context: Word.RequestContext) => {
+      const { contentControls } = context.document;
+      context.load(contentControls, "tag, length");
+      await context.sync();
+      let length = 0;
+      let pos = 0;
+      for (let i = 0, ilen = contentControls.items.length; i < ilen; i += 1) {
+        const { tag } = contentControls.items[i];
+        if (tag.includes("JABREF-CITATION")) {
+          if (tag.substring(16) === "NEW") {
+            pos = length;
+            break;
+          }
+          length += 1;
+        }
+      }
+      return pos;
+    }).catch((error) => {
+      console.log(`Error: ${JSON.stringify(error)}`);
+      if (error instanceof OfficeExtension.Error) {
+        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
+      }
+    });
+  }
+
+  static async getCitationIdToPos(): Promise<Record<string, number> | void> {
+    return Word.run(async (context: Word.RequestContext) => {
+      const { contentControls } = context.document;
+      context.load(contentControls, "tag, length");
+      await context.sync();
+      const citationIdToPos = {};
+      let pos = 0;
+      contentControls.items.forEach((citation) => {
```

Comment:
You quite often need to get all jabref content controls (what a surprise, right?). Thus you could extract
```
      const { contentControls } = context.document;
      context.load(contentControls, "tag, length");
      await context.sync();
      return contentControls.items.filter(isJabRef)
```
to a new method.